src/zenml/stack_stores/local_stack_store.py

```diff
@@ -187,6 +187,58 @@ def register_stack_component(
             component.name,
         )
 
+    def update_stack_component(
+        self,
+        name: str,
+        component_type: StackComponentType,
+        component: StackComponentWrapper,
+    ) -> Dict[str, str]:
+        """Update a stack component.
+
+        Args:
+            name: The original name of the stack component.
+            component_type: The type of the stack component to update.
+            component: The new component to update with.
+
+        Raises:
+            KeyError: If no stack component exists with the given name.
+        """
+        components = self.__store.stack_components[component_type]
+        if name not in components:
+            raise KeyError(
+                f"Unable to update stack component (type: {component_type}) "
+                f"with name '{name}': No existing stack component "
+                f"found with this name."
+            )
+        component_config_path = self._get_stack_component_config_path(
+            component_type=component.type, name=component.name
+        )
+        utils.create_dir_recursive_if_not_exists(
+            os.path.dirname(component_config_path)
+        )
+        utils.write_file_contents_as_string(
+            component_config_path,
+            base64.b64decode(component.config).decode(),
+        )
+        if name != component.name:
+            self._delete_stack_component(component_type, name)
+
+        # add the component to the stack store dict and write it to disk
+        components[component.name] = component.flavor
```

Comment:
I know that you're already checking for the case that a component exists in the CLI methods (and I definitely think it should be there with a CLI error message), however I think we should have it here as well. Users (or us) might still call this function directly in which case we would just overwrite the existing one without noticing. Maybe a `StackComponentExistsError`?